Feat(HomePage): Create Home Page

Kara-Zor-El opened the pull request:
Link to Preview: https://club-website-c1tjlgepg-karas-projects-e1ba5d81.vercel.app
<img width="1712" alt="image" src="https://github.com/user-attachments/assets/5ebc1da2-6145-4675-b6b5-a6a18c367b5c">

## Needed Additions
- [X] Hero banner (#10)
- [x] Section Headers (#20)
- [X] About Us Section (Accidentally put on this branch)
- [X] Upcoming Events (with fallbacks) (Will do so in future PR)
- [X] Latest News / Blogs (#23 for initial UI without content fetching)
- [X] Contact Us (#21)
- [x] Footer (#22)

## Additional notes:
- I have not made this work on mobile yet. I figure I'll do so later and instead make sure it looks good on just desktop for now.
- Latest News / Blogs section doesn't have content fetching
- We will wait for event page in later pr
- Event section not included (and won't be until the events page is merged due to borrowing components)

Kara-Zor-El committed:
Chore: add small commit to be able to turn this into draft pr

Kara-Zor-El committed:
Feat(Section Header): Make section header

Kara-Zor-El committed:
Merge branch 'KaraZorEl-HomePage' into KaraZorEl-SectionHeader

Kara-Zor-El committed:
Feat(Section Header): Make section header (#20)

section header 👍

Kara-Zor-El committed:
Chore(AboutUs): upload image

Kara-Zor-El committed:
Chore(Readme): Fix Readme

Kara-Zor-El committed:
Feat(Config): Add new interface to config for about us

Kara-Zor-El committed:
Feat(AboutUs): add about us page

spotandjake commented on components/SectionHeader.tsx:
```diff
@@ -0,0 +1,28 @@
+interface Props {
+  title: string;
+  backgroundColor?: string;
+  foregroundColor?: string;
```
can we do what we did with the buttons here and just have a style prop which is an enum of prebuilt styles, this way we can verify accessibility.

Kara-Zor-El replied:
sure, on it

Kara-Zor-El replied:
Fixed

spotandjake commented on components/SectionHeader.tsx:
```diff
@@ -0,0 +1,28 @@
+  backgroundColor = 'bg-color-1',
+  foregroundColor = 'bg-dark-gray',
+  textColor = 'text-color-2',
+  location = 'justify-start',
```
I think location would still be seperate from the styles but can we change this to an enum of like `rightAligned | LeftAligned` its far more informative imo.

Kara-Zor-El replied:
Fixed

spotandjake commented on config.d.ts:
```diff
@@ -16,10 +16,17 @@ declare module '*.yaml' {
     pageInfo: PageInfo[];
     bannerInfo: BannerInfo;
     heroIcons: HeroIcon[];
+    aboutUs: AboutUs;
```
Thoughts on changing this to like:
```
interface HomeSection {
  title: string;
  content: string;
  image: string;
  imageAlt: string;
  buttonRoute?: string;
}
```
 Then we could automatically add more sections to the page easily in terms of text content, I know this doesnt take into consideration other section types but I think thats okay, If you want to keep it the way it is thats fine too, I just figure making it a little more open makes things more expandable in the future.

Kara-Zor-El replied:
I like that idea. not 100% sure how it should be implemented in the main page (like if we want to do stuff back to back or grab specific indexes to cherrypick but open to discussion on that. For now that feature is implemented :)

spotandjake commented on components/AboutUs.tsx:
```diff
@@ -0,0 +1,26 @@
+import React from 'react';
```
Aboutus is a bad component, it would be better to make this a generic `TextSection` component and handle all the content from the home section.

Kara-Zor-El replied:
changed

spotandjake commented on tailwind.config.ts:
```diff
@@ -14,11 +14,12 @@ const config: Config = {
         'color-1': '#0f3b26', // Main Green
         'color-2': '#f9f7f7', // NavBar Off White
         'color-3': '#575555', // Aside Gray
+        'dark-gray': '#222', // Dark Gray
```
can you use `color-4` instead, I know its less descriptive but named colors is bad for theme management, colors should be irrespective on content and more about location of use, This entire color pallete needs to be rewritten tbh to use a good pallete with names like `primary-bg` and stuff.

Kara-Zor-El replied:
fixed

spotandjake reviewed (commented):
I know your still working just some feedback on current implementation. Feel free todo things differently if you would like, just some notes on how we can make thigns a bit more templated and re-usable for later.

Kara-Zor-El committed:
Feat(Contact Us): Contact us section

Kara-Zor-El committed:
Feat(ContactUs): make email unkderline and stylize

Kara-Zor-El committed:
Feat(ContactUs): Create Contact us component for main page (#21)

Kara-Zor-El committed:
Fix(ContactUs): Formatting

Kara-Zor-El committed:
Fix(ContactUs): Add new text

Kara-Zor-El committed:
Fix(HeroBanner): Change text

Kara-Zor-El committed:
Fix(ContactUs): Minor adjustments

Kara-Zor-El committed:
Fix(SectionHeader): use enum for where to align it

Kara-Zor-El committed:
Fix(TextSection): Change AboutUs to be a TextSection

Kara-Zor-El committed:
Fix(SectionHeader): Add enum for styling

Kara-Zor-El committed:
Feat(TextSection): allow HomeSection to produce multiple TextSections

Kara-Zor-El committed:
Feat(Footer): Added footer

Kara-Zor-El committed:
Feat(Footer): Added footer (#22)

Kara-Zor-El committed:
FIx(TextSection): Fix CSS naming

Kara-Zor-El committed:
Feat(LatestNews): Add Latest News section to homepage

Kara-Zor-El committed:
Feat(LatestNews): Add Latest News section to homepage (#23)

Kara-Zor-El committed:
Merge branch 'main' into KaraZorEl-HomePage

Kara-Zor-El committed:
Fix(SocialButton): Align center and change padding

Kara-Zor-El committed:
Merge branch 'KaraZorEl-HomePage' of https://github.com/Trent-Computer-Science-Club-Association/ClubWebsite into KaraZorEl-HomePage

spotandjake commented on components/ContactUs.tsx:
I think we should move this into the `Footer`, if we decide against this can we move it from the components folder to a `layout` folder, its not really a component because it wont be reused. (Need to move styles too if we take that route.

Kara-Zor-El replied:
fixed

spotandjake commented on components/ContactUs.tsx:
```diff
@@ -0,0 +1,49 @@
+import Image from '../components/Image';
+import Button, { ButtonType } from '../components/Button';
+import Link from '../components/Link';
+import { Fragment } from 'react';
```
I don't think we need this after

spotandjake commented on components/ContactUs.tsx:
```diff
@@ -0,0 +1,49 @@
+      </div>
+      <div className={styles.right}>
+        <p className={styles.text}>
+          {contactUs.text.split('*contactEmail').map((part, index, array) =>
```
Can we just use markdown here.

spotandjake commented on components/Button.tsx:
Can we modify button to take an `icon` url, this will let us handle the image scaling responsivly, if this doesn't get done for this pr can we open an issue.

Kara-Zor-El replied:
fixed

spotandjake commented on components/ContactUs.tsx:
```diff
@@ -0,0 +1,49 @@
+                key={icon.altText}
+                src={icon.path}
+                alt={icon.altText}
+                width={50}
```
Test this for responsiviety, and maybe take a look at making the button handle the image stuff.

Kara-Zor-El replied:
fixed

spotandjake commented on components/Footer.tsx:
```diff
@@ -0,0 +1,36 @@
+
+export default function Footer() {
+  return (
+    <footer className={styles.Footer}>
```
Let's maybe combine this with the contactus section, if we do that then this can just be the logo and a copyright notice.

If we decide against that can we flip the the content around.

Kara-Zor-El replied:
Hey, did this, doesn't look great. Lets talk more on how to better style the UI here

spotandjake commented on components/Footer.tsx:
```diff
@@ -0,0 +1,36 @@
+    <footer className={styles.Footer}>
+      {/* Left side */}
+      <div className={styles.left}>
+        <Image src='/logo.svg' alt='TCSCA Logo' width={100} height={100} />
```
Why does this not use the logo component?

Kara-Zor-El replied:
Hi, the answer is because I forgot

Kara-Zor-El replied:
fixed now

spotandjake commented on components/Footer.tsx:
```diff
@@ -0,0 +1,36 @@
+        <div className={styles.socialMedia}>
+          {config.heroIcons.map((icon) => (
+            <Button key={icon.altText} type={ButtonType.SOCIAL_DARK}>
+              <Image
```
Same as with the last icon button.

Kara-Zor-El replied:
fixed?

spotandjake commented on components/HeroBanner.tsx:
```diff
@@ -22,7 +22,7 @@ export default function HeroBanner({ imagePath, altText }: Props) {
         <h1>Trent Computer Science Club Association</h1>
         <hr />
-        <div className={styles.tagline}>Where passion meets code</div>
+        <div className={styles.tagline}>Where innovation happens</div>
```
should we take this from the `config`?

Kara-Zor-El replied:
sure

Kara-Zor-El replied:
fixed

spotandjake commented on components/LatestNews.tsx:
Can we move this out of the components section and to a `layout` section, Components are things that are reusable.

Kara-Zor-El replied:
:+1:

spotandjake commented on components/LatestNews.tsx:
```diff
@@ -0,0 +1,33 @@
+//  NOTE: will not take in props for now (later will be able to take in props or just process the config)
+export default function LatestNews() {
+  return (
+    <Link className={styles.latestNews} href='/'>
```
This should be on each item right? can we have a section wrapping this.

spotandjake commented on components/LatestNews.tsx:
```diff
@@ -0,0 +1,33 @@
+export default function LatestNews() {
+  return (
+    <Link className={styles.latestNews} href='/'>
+      <div className={styles.newsItem}>
```
You can make these their own component.

spotandjake commented on components/Link.tsx:
I think we can get rid of this, I can't see a case come up where we need to override the global behaviour of the link element.

Kara-Zor-El replied:
fixed

spotandjake commented on components/SectionHeader.tsx:
This is a layout not a component.

Kara-Zor-El replied:
fixed

spotandjake commented on components/SectionHeader.tsx:
```diff
@@ -0,0 +1,49 @@
+  style = SectionHeaderStyle.BLACK_ON_GREEN,
+  location = SectionLocation.LEFT_ALIGNED,
+}: Props) {
+  let backgroundColor: string;
```
Can you move this to the style logic.

Kara-Zor-El replied:
Fixed

spotandjake commented on components/SectionHeader.tsx:
```diff
@@ -0,0 +1,49 @@
+  return (
+    <div className={`${styles.sectionHeader} ${backgroundColor} ${location}`}>
+      <div className={`${styles.sectionHeaderContainer} ${foregroundColor}`}>
+        <h2 className={`${styles.sectionHeaderTitle} `}>{title}</h2>
```
```suggestion
        <h2 className={styles.sectionHeaderTitle}>{title}</h2>
```

Kara-Zor-El replied:
fixed

spotandjake commented on components/SectionHeader.tsx:
```diff
@@ -0,0 +1,49 @@
+  }
+  return (
+    <div className={`${styles.sectionHeader} ${backgroundColor} ${location}`}>
+      <div className={`${styles.sectionHeaderContainer} ${foregroundColor}`}>
```
I feel like this should maybe be a `span` instead of a div, given it doesn't break up content it just isolates it for styling.

Kara-Zor-El replied:
fixed

spotandjake commented on components/TextSection.tsx:
```diff
@@ -0,0 +1,63 @@
+}: Props | PropsWithButton) {
+  let backgroundColor: string;
+
+  switch (style) {
```
Can we move this over to classes as well.

Kara-Zor-El replied:
fixed

spotandjake commented on components/TextSection.tsx:
```diff
@@ -0,0 +1,63 @@
+      <Image
+        src={imagePath}
+        alt={altText}
+        width={100}
```
This needs to be responsive.

spotandjake commented on components/TextSection.tsx:
```diff
@@ -0,0 +1,63 @@
+}
+
+// props or props with button
+export default function TextTextSection({
```
```suggestion
export default function TextSection({
```

Kara-Zor-El replied:
fixed

spotandjake commented on components/TextSection.tsx:
Move this to a layouts folder (Dont forget the styles).

Kara-Zor-El replied:
fixed

spotandjake commented on config.d.ts:
```diff
@@ -16,10 +16,26 @@ declare module '*.yaml' {
+    text: string;
+    image: string;
+    imageAlt: string;
+    buttonText?: string;
```
Maybe we just use markdown and inject the button with markdown (Need to look into how we convert a markdown button to our special button.

Kara-Zor-El replied:
Hey Jake, cool idea and all, I'd rather not. Just sounds like were gonna have to explain how to use flavoured markdown to a non dev and for what? extra work? Don't really see the point

spotandjake commented on config.d.ts:
```diff
@@ -16,10 +16,26 @@ declare module '*.yaml' {
+  interface ContactUs {
+    text: string;
+  }
+
```
What if we tried a config approach like this, it means everything on the page needs to come from the config but greatly reduces the complexity of changes.
```ts
enum HomeSectionType {
  TextSection = "TextSection"
}
interface HomePageSection {
  sectionType: HomeSectionType;
  sectionHeader: string;
}
interface TextSection implements HomePageSection {
  sectionType: HomeSectionType.TextSection;
  sectionHeader: string;
  text: string;
  // ...etc
}
type HomeSection = TextSection | OtherSection;
```

```yaml
homePage:
  - sectionType: "TextSection"
    text: "this is the text section"
  - sectionType: "NewSection"
    text: "This is the enws stuff except it wouldnt be text it would be a list of news items or a data source, idk how we want to configure that just yet"
```

spotandjake replied:
More context

spotandjake replied:
```tsx
<Section sectionConfig={} even={} key={}/>
```
```tsx
let section: JSX.Element | undefined; // get rid of undefiend case by using an external mapping function
switch (sectionConfig.sectionType) [
  case "textSection":
    section = <TextSection />
    break;
}
if (section == undefined) throw new Error("Impossible"); 
return (
  <section classname={even ? style.even : style.odd}>
    <SectionHeader />
    {section}
  </section>
)
```
```tsx
getSectionElement();
```
```tsx
sectionConfig={}
```

Kara-Zor-El replied:
Hey Jake, can we throw this on the backburner? I understand your approach but this seems like slightly too much config driven development, especially for right now where it might be better to just get everything done and then go back to it later

spotandjake replied:
> Hey Jake, can we throw this on the backburner? I understand your approach but this seems like slightly too much config driven development, especially for right now where it might be better to just get everything done and then go back to it later

We 100% can but I think we should keep it in mind for the future, if we had started with a config driven approach it probably would have been less work in the first place and would make additons earlier (We didn't have things as planned out though, so it made sense to use the approach you used). I think we'll keep it the way you have it and just think about this practice for future pages.

Kara-Zor-El replied:
Changed

spotandjake commented on config.yaml:
```diff
@@ -27,3 +27,14 @@ heroIcons:
     path: /Icons/discord.svg
+homeSections:
+  - title: About Us
+    text: The Trent Computer Science Club Association (TCSCA) is a club that aims to provide a supportive community for computer science students at Trent University. Our mission is to create a welcoming and inclusive environment where students can learn, grow, and have fun while developing their skills and interests. We love Open Source and are always looking for new ways to contribute to the co ...
```
We need to get this content.

Kara-Zor-El replied:
updated

spotandjake commented on pages/index.tsx:
```diff
@@ -13,6 +21,48 @@ export default function Home() {
           <HeroBanner imagePath='logo.svg' altText='TCSCA Logo' />
+          {/* NOTE: we may want to put stuff in seperate sections rather than back to back */}
+          {homeSections.map((section, index) => {
+            const sectionStyle =
```
maybe we just use an `oddSection` and `evenSection` class it might just make things a bit simplier.

Alternatively maybe we just use:
```css
:nth
```

Kara-Zor-El replied:
Did it in an alternative way

spotandjake commented on pages/index.tsx:
```diff
@@ -13,6 +21,48 @@ export default function Home() {
+              index % 2 === 0 ? TextSectionStyle.BLACK : TextSectionStyle.GREEN;
+            return (
+              <>
+                <SectionHeader
```
Maybe we make this into a a `Section` Component which handles matching the seciton to the child, it could also maybe take the `Section` from the config and handle mapping the config to a section.

Kara-Zor-El replied:
Fixed

spotandjake commented on pages/index.tsx:
```diff
@@ -13,6 +21,48 @@ export default function Home() {
+          />
+          <LatestNews />
+          <ContactUs />
+          <Footer />
```
Footer should not be inside main

Kara-Zor-El replied:
Fixed

spotandjake commented on styles/components/ContactUs.module.scss:
```diff
@@ -0,0 +1,49 @@
+  @apply p-5 bg-color-1;
+}
+
+.title {
```
Should title just be an h something?

Kara-Zor-El replied:
Removed contact us anyways

spotandjake commented on styles/components/ContactUs.module.scss:
```diff
@@ -0,0 +1,49 @@
+  @apply px-2;
+}
+
+.text {
```
text should probably just be a p.

Kara-Zor-El replied:
Removed contact us anyways

spotandjake commented on styles/components/LatestNews.module.scss:
```diff
@@ -0,0 +1,38 @@
+      @apply bg-hover-1;
+      @apply shadow-lg;
+      @apply transition-all;
+      @apply duration-300 ease-in-out;
```
Move this to the parent, you change the item on hover, the transition stuff always stays.

Kara-Zor-El replied:
fixed

spotandjake commented on styles/components/LatestNews.module.scss:
```diff
@@ -0,0 +1,38 @@
+    &:hover {
+      @apply bg-hover-1;
+      @apply shadow-lg;
+      @apply transition-all;
```
Can we use the specific properties here, all is slow.

Kara-Zor-El replied:
fixed

spotandjake reviewed (commented):
A few changes, most of them will solve the others though, so I hope its a bit less work then it looks like. But let me know if you need any help.

Kara-Zor-El commented:
~change color-4 to be color-3~

Kara-Zor-El commented:
~remove shaddow for buttons (for title bar)~

Kara-Zor-El committed:
Fix(Footer): Remove Contact us and moved to footer

Kara-Zor-El committed:
Fix(Button): Allow button to take in an Icon

Kara-Zor-El committed:
Fix(Button): Formatting

Kara-Zor-El committed:
Fix(Footer): Use logo component

Kara-Zor-El committed:
Fix(heroBanner): Add tagline to config

Kara-Zor-El committed:
Fix(Formatting)

Kara-Zor-El committed:
Fix(TextSection): spelling mistake

Kara-Zor-El committed:
Fix(Link): Don't override global behavior

Kara-Zor-El committed:
Fix(News): Change news to be layout and make it proper

Kara-Zor-El committed:
Fix(Footer): Make buttons clickable

Kara-Zor-El committed:
FIx(SectionHeader): moving logic and changing div to span

amberwelby commented on components/Button.tsx:
```diff
@@ -16,33 +18,50 @@ interface Props {
+
+  if (props.icon) {
+    return (
+      // also include styles.iconContainer and classes
```
Is this comment a note to self or explaining the section below?

Kara-Zor-El committed:
Fix(SectionHeader): Formatting

Kara-Zor-El committed:
Fix(SectionHeader): Move to style logic

Kara-Zor-El committed:
Fix(index): change component to layout

Kara-Zor-El committed:
Fix(TextSection): Change logic to style

Kara-Zor-El committed:
FIx(LayoutNews): Fix some styling

amberwelby commented on styles/components/TextSection.module.scss:
```diff
@@ -0,0 +1,25 @@
+.TextSection {
+  @apply grid grid-cols-2 gap-4 p-4;
+  // @apply bg-color-4;
```
Assuming this is commented out to be removed so I thought I'd highlight it

amberwelby commented on styles/components/TextSection.module.scss:
```diff
@@ -0,0 +1,25 @@
+  @apply leading-relaxed;
+  @apply text-xl;
+  @apply pr-4;
+  @apply mb-4;
```
Is there a benefit to having each rule on it's own line? Or is it worth saving lines by grouping rules (ie @ apply {all of the colouring rules} @ apply {all of the padding/margins/gaps}

Kara-Zor-El replied:
I mean no real advantage other than saving a few lines to doing it together. Personally I just do whatever makes sense to me in the moment. Changed it tho

amberwelby reviewed (approved):
Just some silly little comments that crossed my train of thought. 
This looks really good Kara

Kara-Zor-El committed:
Fix(TextSection): remove unnecessary commented out area

Kara-Zor-El committed:
Fix(Button): remove unnecessary commented out area

Kara-Zor-El committed:
Fix(TextSection): Chnage # of lines in styling

Kara-Zor-El committed:
FIx(TextSection): reduce classname usage

Kara-Zor-El committed:
Fix(Footer): Outside of main

Kara-Zor-El committed:
Fix(Index): Im just a silly girl who makes mistakes

Kara-Zor-El committed:
Fix(NavBar): Make NavBar button have no hover

Kara-Zor-El committed:
Fix(Color-4): Change to color-3

Kara-Zor-El committed:
Fix(Color): in section headers

Kara-Zor-El committed:
Update config.yaml

Kara-Zor-El committed:
Fix(TextSection): Make a Layout

spotandjake committed:
chore(home): Some quick cleanup

Kara-Zor-El committed:
Fix(TextSection): Fix theming

Kara-Zor-El committed:
Fix(SectionHeader): Set Alignment via scss

Kara-Zor-El committed:
Fix(SectionHeader): formatting

Kara-Zor-El committed:
Feat(Section): Add Section Component

Kara-Zor-El committed:
Fix(TextSection): Make Image size responsive

spotandjake commented on components/Button.tsx:
```diff
@@ -16,33 +18,49 @@ interface Props {
-  label,
-  children,
+interface IconProps extends Props {
+  icon: string;
```
I think we should change this to:
```ts
image: {
  src: string;
  alttext: string;
}
```
my thinking is that it keeps things together.

Also I feel like instead of having a seperate icon button it might make sense to allow for an icon and a label. and if you only give one we just adapt to that situation. (If you wanna get this merged though we could come back to that in another pr).

Kara-Zor-El replied:
fixed

spotandjake commented on components/Button.tsx:
```diff
@@ -16,33 +18,49 @@ interface Props {
+  altText?: never;
 }
 
 interface ChildrenProps extends Props {
```
Can you think of a use case we would accept for a childrenProp button still? the only thing I could see you passing in is bolded text or something which would be better acheieved by just overriding the styling.

Kara-Zor-El replied:
You're right, fixed

spotandjake commented on components/Button.tsx:
```diff
@@ -16,33 +18,49 @@ interface Props {
   className,
   type = ButtonType.NAVBAR,
-  prefetch = false,
+  ...props
```
I am not sure I love what you have done here, but I do understand the complexity of the situation and cant think of an entirely better alternative.

spotandjake commented on components/Button.tsx:
```diff
@@ -16,33 +18,49 @@ interface Props {
 }) => {
   const classes = `${className} ${type} ${styles.button}`;
-  const content = label || children;
+  const content = props.label || props.children || props.icon;
```
I think we should remove `props.icon` here as we do not want the image src somehow ending up as the button content. (I know that physically can't happen but thats more of why)

Kara-Zor-El replied:
Fixed

spotandjake commented on layouts/LatestNews.tsx:
```diff
@@ -0,0 +1,52 @@
+  style?: NewsSectionStyleType;
+}
+
+const LatestNewsLayout: React.FC<LatestNewsLayoutProps> = ({
```
for future note I think a better approach here might to the config approach so we have a `<NewsSection` component that takes something like this in:
```ts
interface NewsItem {
  text: string; // as markdown
  href: string; // Where the learn more would take you
  date: Date;
}

interface NewsSectionProps {
  newsFeed: NewsItem[];
}
// and then its just a for each loop that maps the newsFeed to newsItems
```

I think the approach used here works though so we can keep it the way it is.

Kara-Zor-El replied:
Changed to that

spotandjake commented on layouts/LatestNews.tsx:
```diff
@@ -0,0 +1,52 @@
+  (typeof NewsSectionStyle)[keyof typeof NewsSectionStyle];
+
+interface LatestNewsLayoutProps {
+  children: React.ReactNode;
```
I really discourage taking `children` in react I know it makes things easier to make unique but it also creates way more surface area for bugs or odd combos especially around css like what happens when someone tries to shove an image here, formed data is ussually a lot better. I am fine with the way things are now but I think its an important note for future development.

Kara-Zor-El replied:
Fixed

spotandjake commented on layouts/LatestNews.tsx:
```diff
@@ -0,0 +1,52 @@
+        <p className={styles.date}>{date}</p>
+      </div>
+      <div className={styles.right}>
+        <Button type={ButtonType.LIGHT} className={styles.learnMore}>
```
I think this should take an `href` too, I know we have the outer link which should capture things but the button should still perform the action it says imo.

Kara-Zor-El replied:
Fixed

spotandjake commented on layouts/Section.tsx:
```diff
@@ -0,0 +1,53 @@
+}
+
+const Section: React.FC<SectionProps> = ({ title, index, children }) => {
+  const sectionStyle =
```
can we convert this logic all into a `getStyle` function and then the `getStyle` function can bassically look like:
```ts
const getStyle = (index) {
  if (index % 2) {
    return { sectionStyle: style, newSectionStyle: style }
  } else {
   return // default style
  }
}

// and to call use a destructure
const { sectionStyle, newsSectionStyle } = getStyle()
```
the benefits to this are one its an easy change but it localizes all the logic which helps prevent syncing issues i.e if we wanted to add a third style and have it change to every three its canging one if statement instead of a bunch of conditionals. we need to convert it to a seperate function though because otherwise we have to rely on mutability which is bad.

Kara-Zor-El replied:
Fuxed

spotandjake commented on layouts/Section.tsx:
```diff
@@ -0,0 +1,53 @@
+  const textSectionStyle =
+    index % 2 === 0 ? TextSectionStyle.BLACK : TextSectionStyle.GREEN;
+
+  console.log(index);
```
Can we remove the debug log

Kara-Zor-El replied:
Fixed

spotandjake commented on layouts/Section.tsx:
```diff
@@ -0,0 +1,53 @@
+  return (
+    <section>
+      <SectionHeader title={title} style={sectionStyle} location={location} />
+      {React.isValidElement(children) &&
```
I am not a fan of this logic and this is why the children based approach is bad compared to the strict format.

Can you make this logic into nice `if` statements, you might need to use the function approach to avoid mutability for this, but this is an overuse of ternarys.

can you also open an issue documenting the config driven approach and add a `// TODO(issuenumber): quick description of change` because I want to fix this logic.

Kara-Zor-El replied:
Fixed logic, will circle back around to making issue later

Kara-Zor-El replied:
@spotandjake is the new logic satisfactory?
```ts
const cloneChildWithStyle = (
  child: SectionChild,
  textSectionStyle: TextSectionStyleType,
  newsSectionStyle: NewsSectionStyleType
) => {
  if (React.isValidElement(child)) {
    if (child.type === TextSection) {
      return React.cloneElement(child, { style: textSectionStyle });
    }
    if (child.type === LatestNewsLayout) {
      return React.cloneElement(child, { style: newsSectionStyle });
    }
  }
  return child;
};
```

Kara-Zor-El replied:
changed it again after this to something I think is better

spotandjake commented on layouts/TextSection.tsx:
```diff
@@ -0,0 +1,57 @@
+      <div className={`${styles.TextSection} ${style}`}>
+        <div className={styles.TextAndButton}>
+          <p>{text}</p>
+          {'buttonText' in props && 'link' in props && (
```
can you just ad da comment here explaining what this does/ I think it needs to be one of the `{/* Comment */}` style comments because its in the jsx

Kara-Zor-El replied:
Fixed

spotandjake commented on pages/index.tsx:
```diff
@@ -12,8 +16,38 @@ export default function Home() {
         <main className={styles.MainArea}>
-          <HeroBanner imagePath='logo.svg' altText='TCSCA Logo' />
+          <HeroBanner />
+          {homeSections.map((section, index) => (
```
add a comment above this explaining what this is for pls.

spotandjake commented on pages/index.tsx:
```diff
@@ -12,8 +16,38 @@ export default function Home() {
+              </Section>
+            </section>
+          ))}
+          <Section title='Latest News' index={homeSections.length}>
```
Can you add a comment above this, litterally ust something like `{/* News Section */}` to break up the content from the logic above so we remeber its seperate.

spotandjake commented on pages/index.tsx:
```diff
@@ -12,8 +16,38 @@ export default function Home() {
+            <LatestNews>
+              <NewsItem
+                title='NEWS ABOUT TRENT COMPUTER SCIENCE'
+                date='September 10th, 2024'
```
Date here would be a javascript `Date` object that we format into a readable date (were going to have to deal with timezones eventually, lets at least be prepared for it).

Kara-Zor-El replied:
Fixed (config driven and allows date object)

spotandjake commented on styles/Button.module.scss:
```diff
@@ -2,14 +2,37 @@
   @apply hover:shadow-lg;
 }
 
+.socialBtn,
```
Is there a reason this doesnt just exist under the `.button` class?

Kara-Zor-El replied:
Good catch, fixed

spotandjake commented on styles/Button.module.scss:
```diff
@@ -2,14 +2,37 @@
+  }
+}
+
+.iconContainer {
```
out of curiousity what would happen if this styling was applied to the other button types? (The more dynamic we make the styling, the better. The less settings in our jsx makes things a lot more nimble and consistent).

Kara-Zor-El replied:
Added

spotandjake commented on styles/Button.module.scss:
```diff
@@ -2,14 +2,37 @@
+}
+
+.iconContainer {
+  @apply flex;
```
like I feel like every button being `flex` and `relative` would not cause any problems.

Kara-Zor-El replied:
Added relative but kept flex-inline since it works better imo

spotandjake commented on styles/components/Footer.module.scss:
```diff
@@ -0,0 +1,70 @@
+    }
+  }
+
+  .logo {
```
Shouldn't the logo element be handling a lot of this styling itself?

Kara-Zor-El replied:
removed the stuff that it already handles now

spotandjake commented on styles/components/Footer.module.scss:
```diff
@@ -0,0 +1,70 @@
+    @apply h-10;
+    @apply justify-end;
+
+    img {
```
why are we styling the `img` here shouldn't that be handled by the iconButton?

spotandjake commented on styles/layouts/LatestNews.module.scss:
```diff
@@ -0,0 +1,53 @@
+    @apply text-white;
+    @apply grid grid-cols-1;
+
+    .title {
```
why is there a `title` class? this would probably be better off being converted to an `h3` or something like that.

Kara-Zor-El replied:
fixed using `p` instead of `h3` due to it being a `ReactMarkdown` section now

spotandjake commented on styles/layouts/LatestNews.module.scss:
```diff
@@ -0,0 +1,53 @@
+  }
+}
+
+.blackOnGreen {
```
I don't know how I did not catch this before, but we should change these style names. we name our colors things like `bg-color-3` so we can easily change the page theme without having incorrect names or needing to update names it probably doesnt make sense to link our style names to them either. maybe just name this `primaryStyle`, `secondaryStyle`? I know its a lot less specific but I think it follows the color naming convention.

spotandjake commented on styles/layouts/SectionHeader.module.scss:
```diff
@@ -0,0 +1,42 @@
+  @apply flex-grow-0;
+}
+
+.sectionHeaderTitle {
```
why a `title` class again? wouldn't this be better as a `h` or some sort and we just style the h directly?

Kara-Zor-El replied:
fixed

spotandjake commented on styles/layouts/SectionHeader.module.scss:
```diff
@@ -0,0 +1,42 @@
+  }
+}
+
+.rightAligned {
```
Can we just add a comment above here something like `// Header Alignments`

Kara-Zor-El replied:
fixed

spotandjake commented on styles/layouts/TextSection.module.scss:
```diff
@@ -0,0 +1,44 @@
+  @apply object-center;
+}
+
+.blackOnGreen {
```
same not as I had before about the colors

spotandjake reviewed (commented):
Things mostly look great, I left a few comments just about best practice going forward and the advantages which we can talk about furthur at a later date, a few things are just adding comments, and then a few are just small style changes. Everything mostly looks great, great work Kara.

Kara-Zor-El committed:
Fix(Button): Make image button take image

Kara-Zor-El committed:
Fix(Button): Remove children props

Kara-Zor-El committed:
Fix(HeroBanner): Formatting

Kara-Zor-El committed:
Fix(Button): Remove unneccary variable

Kara-Zor-El committed:
Feat(LatestNews): Refactor news section

Kara-Zor-El committed:
Fix(Section): Update style logic

Kara-Zor-El committed:
Chore(Section): Formatting

Kara-Zor-El committed:
Fix(Section): Turn turinaries into if's

Kara-Zor-El committed:
Chore(TextSection): Document code

Kara-Zor-El committed:
Chore(Index): Add comments

Kara-Zor-El committed:
Fix(Button): Change CSS

Kara-Zor-El committed:
Chore(Button): Move some CSS

Kara-Zor-El committed:
Chore: move some CSS

Kara-Zor-El committed:
Fix(Button): Remove redundant code

Kara-Zor-El committed:
Fix(SectionHeader): fix styling

Kara-Zor-El committed:
Chore(SectionHeader): Add comment for header alignments

Kara-Zor-El committed:
Fix(Styling): Change names to be more dynamic

Kara-Zor-El committed:
Feat(Section): Take Sections from config

Kara-Zor-El committed:
Fix(HomePage): Remove commented out bits

Kara-Zor-El committed:
Fix(Section): Make colors more generic

Kara-Zor-El committed:
Fix(HomePage): Remove unnecessary imports

Kara-Zor-El commented:
Closes #6

Kara-Zor-El committed:
Feat(SectionHeader): Make Rounded

spotandjake commented on components/Button.tsx:
```diff
@@ -16,38 +18,49 @@ interface Props {
 }) => {
   const classes = `${className} ${type} ${styles.button}`;
-  const content = label || children;
+  if (props.image) {
```
one refactor here which will make things better for maintence make this:
```tsx
const content = <>
  {(props.image == undefined ? <></> : <div className={styles.imageWrapper}>
        <Image src={props.image.src} alt={props.image.altText} fill={true} />
      </div>)}
      {props.label == undefined ??<></> : (<span className={styles.buttonText}>{props.label}</span>)}
</>
```

spotandjake commented on components/Button.tsx:
```diff
@@ -16,38 +18,49 @@ interface Props {
   const classes = `${className} ${type} ${styles.button}`;
-  const content = label || children;
+  if (props.image) {
+    return (
```
Then we can get rid of the route this.

spotandjake commented on components/Button.tsx:
```diff
@@ -16,38 +18,49 @@ interface Props {
+    );
+  }
 
   return href ? (
```
and we can just pass the content in down here :) which means one point of maintence.

spotandjake commented on config.d.ts:
```diff
@@ -16,10 +16,56 @@ declare module '*.yaml' {
+    email: string;
+    tagline: string;
+    footer: Footer;
+    HomeSection: HomeSection;
```
why is there a homeSection and a `homeSections`? and why is the capitalization different for this one?

spotandjake commented on layouts/LatestNews.tsx:
```diff
@@ -0,0 +1,82 @@
+const formatDate = (date: Date): string => {
+  const day = date.getDate();
+
+  const getDaySuffix = (day: number): string => {
```
is there no `toLocale` that outputs this right? Maybe check out the `moment` time library from npm it might have what we want. This approach isn't the best for localization imo, which is why i think a time library might be better.

If you can't find something suitable we can keep this logic so maybe only spend like 15 mins looking into `moment` or the default localization api's

spotandjake commented on layouts/Section.tsx:
```diff
@@ -0,0 +1,73 @@
+  (typeof TextSectionStyle)[keyof typeof TextSectionStyle];
+
+interface SectionProps {
+  sectionConfig: typeof HomeSection;
```
Hmms, You might need to make a `types.ts` and import the types in `config.d.ts` from there, I don't like how you have to say `typeof HomeSection` here im pretty sure it means something different. Can you just verify this is a strict typecheck, and if not make that change.

spotandjake commented on layouts/Section.tsx:
```diff
@@ -0,0 +1,73 @@
+  index: number;
+}
+
+const getStyle = (index: number) => {
```
this looks great.

spotandjake commented on layouts/Section.tsx:
```diff
@@ -0,0 +1,73 @@
+  const { sectionStyle, location, newsSectionStyle, textSectionStyle } =
+    getStyle(index);
+
+  let content: React.ReactNode;
```
can we break this out into a `getContent` function, this would avoid the mutability, which is a good practice and honestly makes the code a lot better. also can we use a `switch` statement instead of `if` statements.

spotandjake commented on pages/index.tsx:
```diff
@@ -1,9 +1,11 @@
 import EventBanner from '../components/EventBanner';
 import HeroBanner from '../components/HeroBanner';
+import Footer from '../components/Footer';
+import Section from '../layouts/Section';
```
I think we should rename this to `HomeSection` jus tso we dont run into any naming issues between our section and the html version.

spotandjake reviewed (approved):
Alright a few final nits but once those are fixed, you have my approval I don't think I need to rereview unless you want me to. Great work on the config driven homepage.

Kara-Zor-El committed:
Fix(Various Changes): Fix a few things

- Refactor code in button
- Remove unnecessary section
- Better exports
- Use Moment.js for timezone
- change name of Section to HomeSection
- unnecessary keyof typeof stuff removed
